mono/utils/w32handle.c

```diff
@@ -1346,20 +1346,31 @@ mono_w32handle_wait_multiple (gpointer *handles, gsize nhandles, gboolean waital
 
 				if (i < lowest)
 					lowest = i;
+
+				if (!waitall)
```

Comment:
Realise I should have added a comment about that. That's the behaviour I see an .NET. But I find it a bit strange to not take as many as possible while at it. I can change it back.